### ISSUE TITLE
Automatically update Theos submodules as any of them gets updated

### ISSUE THREAD

**Issue by PoomSmart** [theos/theos#604]
What happens?
-------------
We have GitHub action to update Theos submodules in this repository to the latest. However, you have to trigger the workflow manually. There's room of improvement where we can use GitHub Actions to synchronize changes as per [this Stackoverflow link](https://stackoverflow.com/questions/64407333/using-github-actions-to-automatically-update-the-repos-submodules). It will be more seamless this way.

What were you expecting to happen?
----------------------------------
As soon as any submodules got updated (master branch), this Theos repository will reflect to the changes automatically.

Tasks
- [ ] Add the sync job to the updating submodules workflow in this repository (Theos)
- [ ] Create a workflow with a notify job for https://github.com/theos/dm.pl
- [ ] Create a workflow with a notify job for https://github.com/theos/headers
- [ ] Create a workflow with a notify job for https://github.com/theos/lib
- [ ] Create a workflow with a notify job for https://github.com/theos/logos
- [ ] Create a workflow with a notify job for https://github.com/theos/nic
- [ ] Create a workflow with a notify job for https://github.com/theos/templates

**Comment by kabiroberai** [comment]
What if we used an organization-wide webhook to trigger the `workflow_dispatch` or `repository_dispatch` event? That would make it so we don't need to manually configure each submodule.

**Comment by PoomSmart** [comment]
@kabiroberai I will take a look on that.

**Comment by PoomSmart** [comment]
@kabiroberai Can you elaborate about that solution?

**Comment by kabiroberai** [comment]
tl;dr we could make an org-wide webhook (I'll do that if you don't have access to the webhooks config settings) which sends a POST request to a desired endpoint whenever there's a push event. We could set it up to POST to a serverless applet that we create (Heroku lets us do that for free), and the applet could in turn make a POST to the GitHub API to [trigger](https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event) workflow_dispatch on theos/theos.

Webhook on theos org --POST--> Heroku applet --POST--> workflow_dispatch on theos/theos repo

**Comment by kabiroberai** [comment]
Update: Heroku isn't traditionally serverless, but something like AWS Lambda, which is, should work too. Since we already have some infrastructure running on Netlify we can utilize its functions feature which makes it really simple to integrate with Lambda.

**Comment by PoomSmart** [comment]
@kabiroberai I see that you are already into the idea plus I do not have advanced access to the repository anyway, you can proceed.

**Comment by kabiroberai** [comment]
Done! The function is currently on a private Netlify site but I'll move it to https://theos.dev once we migrate that to Netlify.
